src/transport_performance/gtfs/validation.py

```diff
@@ -601,3 +655,803 @@ def get_route_modes(self):
+        orientation : str, optional
+            The orientation of the bar plot ("v" or "h"),
+            by default "v"
+        plotly_kwargs : dict, optional
+            Kwargs to pass to fig.update_layout() for
+            additional plot customisation,
+            by default {}
+        return_html : bool, optional
+            Whether or not to return a html string,
+            by default False
+        width : int, optional
+            The width of the plot (in pixels), by default 2000
+        height : int, optional
+            The height of the plot (in pixels), by default 800
+        xlabel : str, optional
+            he label for the x axis.
+            If left empty, the column name will be used,
+            by default None
+        ylabel : str, optional
+            he label for the y axis.
+            If left empty, the column name will be used,
+            by default None
+        save_html : bool, optional
+            Whether or not to save the plot as a html file,
+            by default False
+        save_image : bool, optional
+            Whether or not to save the plot as a PNG,
+            by default False
+        out_dir : Union[pathlib.Path, str], optional
+            The directory to save the plot into. If a file extension is added
+            to this directory, it won't be cleaned. Whatever is passed as the
+            out dir will be used as the parent directory of the save, leaving
+            the responsibility on the user to specify the correct path.,
+            by default os.path.join("outputs", "gtfs")
+        img_type : str, optional
+            The type of the image to be saved. E.g, .svg or .jpeg.,
+            by defauly "png"
+
+        Returns
+        -------
+        Union[PlotlyFigure, str]
+            Returns either a HTML string or the plotly figure
+
+        Raises
+        ------
+        AttributeError
+            Raise an error if the daily_route_summary table is yet to be #
+            created.
+
+        """
+        # defensive checks
+        if "daily_route_summary" not in self.__dir__():
+            raise AttributeError(
+                "The daily_route_summary table could not be found."
+                " Did you forget to call '.summarise_routes()' first?"
+            )
+
+        # plot the summary
+        target_col = f"route_count_{target_summary}"
+        plot = self._plot_summary(
+            summary_df=self.daily_route_summary,
+            target_column=target_col,
+            day_column="day",
+            orientation=orientation,
+            plotly_kwargs=plotly_kwargs,
+            return_html=return_html,
+            width=width,
+            height=height,
+            xlabel=xlabel,
+            ylabel=ylabel,
+            save_html=save_html,
+            out_dir=out_dir,
+            save_image=save_image,
+            img_type=img_type,
+        )
+        return plot
+
+    def plot_trip_summary(
```

Comment:
I feel we can generalise plot summary to cope with the behavioural requirements for plot_route_summary and plot_trip_summary. Their behaviours are consistent for different dataframes, right? This would save us approx 300 lines in this module and more again in testing.

the suggestion is to remove plot_trip_summary and plot_route_summary in favour of a plot_summary function that can handle both of their needs.